lib/base/json.cpp

```diff
@@ -372,19 +372,11 @@ void JsonEncoder<prettyPrint>::NumberFloat(double value)
 	if (value < 0) {
 		long long i = value;
 
-		if (i == value) {
-			AppendJson(i);
-		} else {
-			AppendJson(value);
-		}
+		AppendJson(i);
 	} else {
 		unsigned long long i = value;
 
-		if (i == value) {
-			AppendJson(i);
-		} else {
-			AppendJson(value);
-		}
+		AppendJson(i);
 	}
```

Comment:
That diff changes the behavior. This actually checks if the number is representable as (unsigned) long long, i.e. has no decimal places and is in range for this data type and in this case writes it to JSON without the `.0` suffix.

Reply:
I don't think it will ever enter the `else` branch. After all, comparing `(0 == 0.0)` results always true.

Reply:
If value is something like `4.2`, it will be reached.